assisted-service-operator v0.0.1

djzager opened the pull request:
Thanks submitting your Operator. Please check below list before you create your Pull Request.

### New Submissions

* [x] Does your operator have [nested directory structure](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#create-a-bundle)?
* [x] Have you selected the Project *Community Operator Submissions* in your PR on the right-hand menu bar?
* [x] Are you familiar with our [contribution guidelines](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md)?
* [x] Have you [packaged and deployed](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md) your Operator for Operator Framework?
* [x] Have you tested your Operator with all Custom Resource Definitions?
* [x] Have you tested your Operator in all supported [installation modes](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#operator-metadata)?
* [x] Have you considered whether you want use [semantic versioning order](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#updating-your-existing-operator)?
* [x] Is your submission [signed](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#sign-your-work)?
* [x] Is operator [icon](https://github.com/operator-framework/community-operators/blob/master/docs/contributing.md#operator-icon) set?

### Your submission should not

* [x] Modify more than one operator
* [x] Modify an Operator you don't own
* [x] Rename an operator - please remove and add with a different name instead
* [x] Submit operators to both `upstream-community-operators` and `community-operators` at once
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description about the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/operator-framework/community-operators/blob/master/docs/packaging-required-fields.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [ ] Example templates for all Custom Resource Definitions intended to be used
* [ ] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

mvalarh commented:
/ok-to-test

mvalarh commented:
/lgtm

djzager commented:
/hold

djzager commented:
/unhold

There are currently 3 issues that prevent this PR from passing the `ci/prow/deploy-operator-on-openshift`:

1) Requiring the `ClusterDeployment` CRD that is provided by Hive. It appears that, when looking through the logs, the default operator sources are disabled so there is no way for OLM to resolve this dependency; even though Hive is available in the community-operators repo.
2) Requiring the `BareMetalHost` CRD that is provided by [baremetal-operator](https://github.com/openshift/baremetal-operator). There is no community-operator for this so even if :point_up: weren't an issue, this would be breaking.
3) Requiring (2) PersistentVolumeClaims: `postgres-pv-claim` and `bucket-pv-claim`.

The conversation has been started on how to address 3, but I don't think that should stop this PR from being accepted as it is now.

J0zi commented:
@djzager, @flaper87
No problem with dependencies, we can merge it. We will test dependencies in the future, task is in backlog.
The only thing, which should be fixed is how operator "looks". Please paste your CSV to the [preview](https://operatorhub.io/preview) and fill `CONTAINER IMAGE` and `CREATED AT` on the right panel. And also CRD at the bottom. Thanks.
When ready, press `Ready for review`.

flaper87 commented on community-operators/assisted-service-operator/0.0.1/assisted-service-operator.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,516 @@
+    operators.operatorframework.io/builder: operator-sdk-v1.4.2
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
+    repository: https://github.com/openshift/assisted-service
+    support: https://github.com/openshift/assisted-service/issues/new
```
@djzager Adding the following should do the trick:

```
    createdAt: '2021-03-19T09:49:00Z'
    containerImage: quay.io/ocpmetal/assisted-service-operator-bundle:73440529c68424ea318362ded9874042a9d2de8e
```

flaper87 commented on community-operators/assisted-service-operator/0.0.1/assisted-service-operator.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,516 @@
+    - kind: Agent
+      name: agents.adi.io.my.domain
+      version: v1alpha1
+    - kind: InstallEnv
```
```displayName: InstallEnv```

flaper87 commented on community-operators/assisted-service-operator/0.0.1/assisted-service-operator.clusterserviceversion.yaml:
```diff
@@ -0,0 +1,516 @@
+      name: baremetalhosts.metal3.io
+      version: v1alpha1
+    owned:
+    - kind: Agent
```
```displayName: Agent```

flaper87 reviewed (commented):
Commented with the changes required to address @J0zi comments. Unfortunately, I cannot apply them myself or do a proper GH suggestion. Here's the screenshot of what it looks like after adding those lines:

![DeepinScreenshot_select-area_20210319104509](https://user-images.githubusercontent.com/13816/111768821-50ff8500-88a0-11eb-974e-d1f28579046b.png)

flaper87 reviewed (commented):
@djzager awesome, thanks for the update! :+1:

djzager commented:
/retest

J0zi commented:
@djzager, please check what was changed. It is the 5-th push tu the PR and first time Openshift deploy has failed. Seems that 4 other commits were able to be deployed.

From the [log](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_community-operators/3329/pull-ci-operator-framework-community-operators-master-deploy-operator-on-openshift/1373796753379168256#1:build-log.txt%3A2446), I see `CatalogSourcesUnhealthy` bur `READY`.

Feel free to click `Ready for review` when ready.

djzager commented:
@J0zi 

> Is there some dependency on another operator?

Yes. We require the `ClusterDeployment` CRD provided by Hive (also in community-operators) but it appears the default operator sources are [disabled in the test](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_community-operators/3329/pull-ci-operator-framework-community-operators-master-deploy-operator-on-openshift/1373796753379168256#1:build-log.txt%3A2055):

```
TASK [deploy_olm_operator_openshift_upstream : Create the namespace] ***********
task path: /tmp/playbooks2/operator-test-playbooks/upstream/roles/deploy_olm_operator_openshift_upstream/tasks/main.yml:199
changed: [localhost] => changed=true 
  cmd: '/tmp/oc-hmgMOTr8y3iRbuZ5wl0b/bin/oc patch OperatorHub cluster --type json -p ''[{"op": "add", "path": "/spec/disableAllDefaultSources", "value": true}]'''
  delta: '0:00:00.498227'
  end: '2021-03-22 01:26:34.336072'
  rc: 0
  start: '2021-03-22 01:26:33.837845'
  stderr: ''
  stderr_lines: <omitted>
  stdout: operatorhub.config.openshift.io/cluster patched
  stdout_lines: <omitted>
```

Which would explain why ours would be [ready](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_community-operators/3329/pull-ci-operator-framework-community-operators-master-deploy-operator-on-openshift/1373796753379168256#1:build-log.txt%3A2187) (and ready right before it starts waiting for the [assisted-service pods](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_community-operators/3329/pull-ci-operator-framework-community-operators-master-deploy-operator-on-openshift/1373796753379168256#1:build-log.txt%3A2523)):

```
TASK [deploy_olm_operator_openshift_upstream : Wait for the catalogsource to be ready] ***
task path: /tmp/playbooks2/operator-test-playbooks/upstream/roles/deploy_olm_operator_openshift_upstream/tasks/main.yml:225
changed: [localhost] => changed=true 
  attempts: 2
  cmd: /tmp/oc-hmgMOTr8y3iRbuZ5wl0b/bin/oc get catalogsource test-operators-ocs -n openshift-marketplace -o=yaml | grep 'lastObservedState:' | grep READY
  delta: '0:00:00.392465'
  end: '2021-03-22 01:26:58.696711'
  rc: 0
  start: '2021-03-22 01:26:58.304246'
  stderr: ''
  stderr_lines: <omitted>
  stdout: '    lastObservedState: READY'
  stdout_lines: <omitted>
```

And why the catalog-operator seems to be complaining about [unsatisfied dependenies](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_community-operators/3329/pull-ci-operator-framework-community-operators-master-deploy-operator-on-openshift/1373796753379168256#1:build-log.txt%3A2851)

```
I0322 01:34:37.483493       1 event.go:278] Event(v1.ObjectReference{Kind:"Namespace", Namespace:"", Name:"assisted-installer", UID:"35c59cc5-741f-47d7-a156-24582876277d", APIVersion:"v1", ResourceVersion:"28734", FieldPath:""}): type: 'Warning' reason: 'ResolutionFailed' constraints not satisfiable: subscription assisted-service-operator requires test-operators-ocs/openshift-marketplace/optest/assisted-service-operator.v0.0.1, subscription assisted-service-operator exists, test-operators-ocs/openshift-marketplace/optest/assisted-service-operator.v0.0.1 has a dependency without any candidates to satisfy it
``` 

What do you think of my analysis? Is there something else I need to be doing?

J0zi commented:
@djzager, we do not support dependencies yet, please note in a next PR. Merging.
